lib/controllers/auth/signup_controller.dart

```diff
@@ -24,24 +44,63 @@ class SignupController extends GetxController {
         'password': passwordController.value.text
       });
  
-      http.Response response = await http.post(url, body: body,  headers: headers);
+      http.Response response = await http.post(url, body: body, headers: headers);
+      
       if (response.statusCode == 201) {
+        // Automatically log in the user after successful signup
+        await _autoLogin();
+      } else {
+        // Handle signup error
+        isLoading.value = false;
+        throw jsonDecode(response.body) ?? 'Unknown Error occurred';
+      }
+    } catch (error) {
+      // Handle any exceptions
+      isLoading.value = false;
+      _showErrorDialog(error.toString());
+    }
+  }
+
+  Future<void> _autoLogin() async {
+    try {
+      // Set login controller's text fields
+      _loginController.usernameController.text = usernameController.text;
+      _loginController.passwordController.text = passwordController.text;
+
+      // Use existing login function
+      bool loginSuccess = await _loginController.loginFunction();
+
+      if (loginSuccess) {
+        // Clear controllers
         usernameController.clear();
         passwordController.clear();
-        Get.off(const AuthScreen());
-      }
-      else {
-        throw jsonDecode(response.body) ?? 'Unknown Error occured' ;
+        confirmPasswordController.clear();
+
+        // Navigate to HomePage
+        Get.offAll(() => const HomePage());
+      } else {
+        // If login fails after signup
+        isLoading.value = false;
+        _showErrorDialog('Signup successful, but automatic login failed');
```

Comment:
Can you change the "automatic login" part to something else, feels a bit weird. You don't have to tell the user that an error occurred. Would be fine if you just put
"Signup successful".